build/make/k8s-component.mk

```diff
@@ -109,13 +109,18 @@ ${BINARY_HELM}: $(UTILITY_BIN_PATH) ## Download helm locally if necessary.
 component-generate: ${K8S_RESOURCE_TEMP_FOLDER} ## Generate the component yaml resource.
 	@echo "Generating temporary K8s component resource: ${K8S_RESOURCE_COMPONENT}"
 	@if [[ ${STAGE} == "development" ]]; then \
-		sed "s|NAMESPACE|$(K8S_HELM_ARTIFACT_NAMESPACE)|g" "${K8S_RESOURCE_COMPONENT_CR_TEMPLATE_YAML}" | sed "s|NAME|$(ARTIFACT_ID)|g"  | sed "s|VERSION|$(DEV_VERSION)|g" > "${K8S_RESOURCE_COMPONENT}"; \
+		sed "s|namespace: NAMESPACE|namespace: $(K8S_HELM_ARTIFACT_NAMESPACE)|g" "${K8S_RESOURCE_COMPONENT_CR_TEMPLATE_YAML}" | sed "s|name: NAME|name: $(ARTIFACT_ID)|g"  | sed "s|VERSION|$(DEV_VERSION)|g" > "${K8S_RESOURCE_COMPONENT}"; \
 	else \
-		sed "s|NAMESPACE|$(K8S_HELM_ARTIFACT_NAMESPACE)|g" "${K8S_RESOURCE_COMPONENT_CR_TEMPLATE_YAML}" | sed "s|NAME|$(ARTIFACT_ID)|g"  | sed "s|VERSION|$(VERSION)|g" > "${K8S_RESOURCE_COMPONENT}"; \
+		sed "s|namespace: NAMESPACE|namespace: $(K8S_HELM_ARTIFACT_NAMESPACE)|g" "${K8S_RESOURCE_COMPONENT_CR_TEMPLATE_YAML}" | sed "s|name: NAME|name: $(ARTIFACT_ID)|g"  | sed "s|VERSION|$(VERSION)|g" > "${K8S_RESOURCE_COMPONENT}"; \
+	fi; \
+	if [[ -n "${COMPONENT_DEPLOY_NAMESPACE}" ]]; then \
+  		sed -i "s|DEPLOY_NAMESPACE|$(COMPONENT_DEPLOY_NAMESPACE)|g" "${K8S_RESOURCE_COMPONENT}"; \
+  	else \
+  	  	sed -i "s|deployNamespace: DEPLOY_NAMESPACE||g" "${K8S_RESOURCE_COMPONENT}"; \
```

Comment:
A second thing. With `yq` it is possible to add new fields to a document. Did you chose this condition-branching to reference the `deployNamespace` field in the original `k8s-component.tpl` template or to have an easy way to `sed`-remove the line?

The alternative would be a bit simpler:
1. Have no `deployNamespace` field in the template
2. add the field with `yq` only if `COMPONENT_DEPLOY_NAMESPACE` is set

The whole block could look like this:

```bash
	if [[ -n "${COMPONENT_DEPLOY_NAMESPACE}" ]]; then \
  		yq -i '.spec.deployNamespace = "$(COMPONENT_DEPLOY_NAMESPACE)"'  "${K8S_RESOURCE_COMPONENT}" ; \
	fi
```

In turn, this line must be removed from `k8s-component.tpl`:11
```
  deployNamespace: DEPLOY_NAMESPACE

What do you think about this?
```